violetplugin-componentdiagram/src/main/java/com/horstmann/violet/product/diagram/component/ComponentDiagramGraph.java

```diff
@@ -0,0 +1,70 @@
+/*
+ Violet - A program for editing UML diagrams.
+
+ Copyright (C) 2007 Cay S. Horstmann (http://horstmann.com)
+ Alexandre de Pellegrin (http://alexdp.free.fr);
+
+ This program is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program; if not, write to the Free Software
+ Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+package com.horstmann.violet.product.diagram.component;
+
+import com.horstmann.violet.product.diagram.abstracts.AbstractGraph;
+import com.horstmann.violet.product.diagram.abstracts.edge.IEdge;
+import com.horstmann.violet.product.diagram.abstracts.node.INode;
+import com.horstmann.violet.product.diagram.classes.edge.DependencyEdge;
+import com.horstmann.violet.product.diagram.classes.edge.InterfaceInheritanceEdge;
+import com.horstmann.violet.product.diagram.classes.node.BallAndSocketNode;
+import com.horstmann.violet.product.diagram.classes.node.ClassNode;
+import com.horstmann.violet.product.diagram.classes.node.PackageNode;
+import com.horstmann.violet.product.diagram.common.edge.NoteEdge;
+import com.horstmann.violet.product.diagram.common.node.NoteNode;
+import com.horstmann.violet.product.diagram.component.node.ComponentNode;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * A UML component diagram.
+ */
+public class ComponentDiagramGraph extends AbstractGraph {
```

Comment:
Bad formatting. The bracket { should be under the class/method header. Correct it everywhere.